projects/scion/workbench-application-platform/src/lib/core/qualifier-patcher.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2018-2019 Swiss Federal Railways
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ *  SPDX-License-Identifier: EPL-2.0
+ */
+
+import { NilQualifier, Qualifier } from '@scion/workbench-application-platform.api';
+
+/**
+ * Returns a copy of the given intent qualifier with all its wildcard qualifier values
+ * replaced with values of the given capability qualifier, if any.
+ * Also, if the intent qualifier specifies a wildcard key, it is merged with the capability qualifier.
+ *
+ * @param intentQualifier
+ *        qualifier for an intent as specified in the manifest, may contain wildcards as qualifier key (*)
+ *        and/or qualifier value (* or ?).
+ * @param capabilityQualifier
+ *        qualifier for a capability as specified in the manifest, may contain wildcards (* or ?) as qualifier value;
+ *        if `null`, {NilQualifier} is used.
+ */
+export function patchQualifier(intentQualifier: Qualifier, capabilityQualifier: Qualifier): Qualifier {
+  const _capabilityQualifier = capabilityQualifier || NilQualifier;
+  const _intentQualifier = intentQualifier || NilQualifier;
+
+  // Create a working copy of the intent qualifier
+  const _patchedQualifier: Qualifier = Object.assign({}, _intentQualifier);
```

Comment:
- The '_' character is not necessary here.
- Use destructuring instead of `Object.assign()`:
```const patchedQualifier: Qualifier = {...intentQualifier};```

Reply:
One of the qualifiers could be null or undefined, that's why they where assigned to an underscored variable. The function has been rewritten to avoid these variables